### PR TITLE
[sd-vote-boost-twavp-v3] Add new strategy

### PR DESCRIPTION
**Pull request by pierremarsotlyon1**
Fixes # .

Changes proposed in this pull request:
- add a new Stake DAO strategy to compute sdTkn voting power

### CODE REVIEW

**Comment by ChaituVR** [comment]
whats changing from v2 to v3?

**Comment by pierremarsotlyon1** [comment]
> whats changing from v2 to v3?

* query the balance of sdCRV all the pools I pass as parameters and sum it up
* Instead of taking the whole balance of the locker, I use the gauge total supply because it's minus the sdcrv in the pools
* If the address that's voting is the "botAddress", you allocate the sdcrv pools vp
* Else, as usual

File: src/strategies/sd-vote-boost-twavp-v3/index.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
+import { multicall } from '../../utils';
+import { BigNumber } from '@ethersproject/bignumber';
+import { formatUnits } from '@ethersproject/units';
+
+export const author = 'clement-ux';
```
Comment (ChaituVR) [inline src/strategies/sd-vote-boost-twavp-v3/index.ts:5]:
```suggestion
export const author = 'pierremarsotlyon1';
```

File: src/strategies/sd-vote-boost-twavp-v3/index.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  // Maximum of 5 multicall!
+  if (options.sampleStep > 5) {
+    throw new Error('maximum of 5 call');
+  }
+
+  // Maximum of 20 whitelisted address
+  if (options.whiteListedAddress.length > 20) {
+    throw new Error('maximum of 20 whitelisted address');
+  }
+
```
Comment (ChaituVR) [inline src/strategies/sd-vote-boost-twavp-v3/index.ts:33]:
```suggestion
  // Maximum of 20 whitelisted address
  if (options.whiteListedAddress.length > 20) {
    throw new Error('maximum of 20 whitelisted address');
  }
  
  if (options.pools.length > 500) {
    throw new Error('maximum of 500 pools');
  }


```

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**